src/proxy/http/HttpTunnel.cc

```diff
@@ -157,8 +157,9 @@ ChunkedHandler::read_size()
           }
         } else {
           // We are done parsing size
-          if (num_digits == 0 || running_sum < 0) {
-            // Bogus chunk size
+          if ((num_digits == 0 || running_sum < 0) ||                               /* Bogus chunk size */
+              (!ParseRules::is_cr(*tmp) && !ParseRules::is_ws(*tmp) && *tmp != ';') /* Unexpected character */
+          ) {
```

Comment:
Taking a quick look at this, it seems like it would accept "100 junk  \n" as the chunk size of a 256 byte chunk.

Reply:
Yeah, a malformed chunk extension would pass the check. I ignored almost everything that follows a chunk size, but it should not allow reading random data as a chunk size.

Reply:
Just making sure I understand: why is `;` an expected character? I don't ever recall seeing a `;` after the chunk header size.

Reply:
Oh, should we also not be surprised by a LF? We seem to be lenient on that elsewhere in this loop.

Reply:
The semicolon is for `chunk-ext`.
```
  chunk          = chunk-size [ chunk-ext ] CRLF
                   chunk-data CRLF

  chunk-ext      = *( BWS ";" BWS chunk-ext-name
                      [ BWS "=" BWS chunk-ext-val ] )
```
https://www.rfc-editor.org/rfc/rfc9112#name-chunk-extensions

`BWS` is  Bad WhiteSpace, and it's basically optional whitespace.
https://www.rfc-editor.org/rfc/rfc9110.html#section-5.6.3-5

As for LF, I forgot to loosen the rule here. I pushed a commit to allow LF.

Reply:
Sounds good. Thank you for the explanation.